src/nunavut/lang/c/support/serialization.j2

```diff
@@ -161,26 +177,48 @@ static inline void nunavutCopyBits(void* const dst,
     {{ assert('src != NULL') }}
     {{ assert('dst != NULL') }}
     {{ assert('src != dst') }}
-    if ((0U == (src_offset_bits % 8U)) && (0U == (dst_offset_bits % 8U)))  // Aligned copy, optimized, most common case.
+
+    if ((0U == (src_offset_bits & CHAR_SHIFT_MASK)) && (0U == (dst_offset_bits & CHAR_SHIFT_MASK)))  // Aligned copy, optimized, most common case.
     {
-        const {{ typename_unsigned_length }} length_bytes = ({{ typename_unsigned_length }})(length_bits / 8U);
-        // Intentional violation of MISRA: Pointer arithmetics. This is done to remove the API constraint that
-        // offsets be under 8 bits. Fewer constraints reduce the chance of API misuse.
-        const uint8_t* const psrc = (src_offset_bits / 8U) + (const uint8_t*) src;  // NOSONAR NOLINT
-        uint8_t*       const pdst = (dst_offset_bits / 8U) +       (uint8_t*) dst;  // NOSONAR NOLINT
-        (void) memmove(pdst, psrc, length_bytes);
-        const uint8_t length_mod = (uint8_t)(length_bits % 8U);
-        if (0U != length_mod)  // If the length is unaligned, the last byte requires special treatment.
+        const char* last_src;
+        char*       last_dst;
+        unsigned    mask, length_mod;
+        const {{ typename_unsigned_length }} length_chars = ({{ typename_unsigned_length }})(length_bits >> CHAR_SHIFT);
+        const char* const psrc    = (src_offset_bits >> CHAR_SHIFT) + (const char*)src;
+        char*       const pdst    = (dst_offset_bits >> CHAR_SHIFT) + (      char*)dst;
+        (void)memmove(pdst, psrc, length_chars);
+        length_mod = length_bits & CHAR_SHIFT_MASK;
+        if (length_mod == 0)
+            return;
+        last_src  = psrc + length_chars;
+        last_dst  = pdst + length_chars;
+        mask      = (unsigned)((1U << length_mod) - 1U);
+        *last_dst = (*last_dst & (unsigned)(~mask)) | (*last_src & mask);
```

Comment:
There are a ton of compiler warnings/errors for implicit sign conversions. This line generates just one of these. Please be sure you are compiling with all warnings-as-errors.